### PR TITLE
refactor(core): refactor axon start

### PR DESCRIPTION
**Pull request by peterzhb**
**What this PR does / why we need it**:

This PR refactors the axon start function to multiple function blocks.

**Which issue(s) this PR fixes**:

Fixes #

**Which docs this PR relation**:

Ref #

**Which toolchain this PR adaption**:

No Breaking Change

**Special notes for your reviewer**:

NIL

**CI Description**

| CI Name                                   | Description                                                     |
| ----------------------------------------- | --------------------------------------------------------------- |
| *Chaos CI*                                | Test the liveness and robustness of Axon under terrible network condition    |
| *Cargo Clippy*                            | Run `cargo clippy --all --all-targets --all-features`      |
| *Coverage Test*                           | Get the unit test coverage report                             |
| *E2E Test*                                | Run end-to-end test to check interfaces                         |
| *Code Format*                             | Run `cargo +nightly fmt --all -- --check` and `cargo sort -gwc`     |
| *Web3 Compatible Test*                    | Test the Web3 compatibility of Axon                               |
| *v3 Core Test*                            | Run the compatibility tests provided by Uniswap V3             |
| *OCT 1-5 \| 6-10 \| 11 \| 12-15 \| 16-19* | Run the compatibility tests provided by OpenZeppelin           |

**CI Usage**

> Check the CI you want to run below, and then comment `/run-ci`.

**CI Switch**

- [ ] Chaos CI
- [x] Cargo Clippy
- [ ] Coverage Test
- [x] E2E Tests
- [x] Code Format
- [x] Unit Tests
- [x] Web3 Compatible Tests
- [ ] OCT 1-5 And 12-15
- [ ] OCT 6-10
- [ ] OCT 11
- [ ] OCT 16-19
- [ ] v3 Core Tests

### CODE REVIEW

File: devtools/chain/config.toml [hunk 1]
```diff
@@ -33,7 +33,7 @@ overlord_gap = 5
 sync_txs_chunk_size = 5000
 
 [[network.bootstraps]]
-multi_address = "/ip4/127.0.0.1/tcp/8001/p2p/QmNk6bBwkLPuqnsrtxpp819XLZY3ymgjs3p1nKtxBVgqxj"
+multi_address = "/ip4/172.17.0.1/tcp/10000/p2p/QmNk6bBwkLPuqnsrtxpp819XLZY3ymgjs3p1nKtxBVgqxj"
```
Comment (KaoImin) [inline devtools/chain/config.toml]: Rollback this

File: devtools/chain/genesis_single_node.json
Comment (KaoImin) [inline devtools/chain/genesis_single_node.json:1]: Why change genesis?

**Review by KaoImin — CHANGES_REQUESTED** [review]

File: devtools/chain/genesis_single_node.json
Comment (peterzhb) [reply devtools/chain/genesis_single_node.json:1]: probably due to rebase issue

**Review by driftluo — DISMISSED** [review]

**Comment by KaoImin** [comment]
Please rebase the main branch @peterzhb

**Review by KaoImin — DISMISSED** [review]

**Review by KaoImin — APPROVED** [review]

**Pull request merged**